src/Microsoft.ComponentDetection.Detectors/pip/IPyPiClient.cs

```diff
@@ -154,7 +155,13 @@ public async Task<IList<PipDependencySpecification>> FetchPackageDependenciesAsy
             {
                 using var r = new PypiRetryTelemetryRecord { Name = spec.Name, DependencySpecifiers = spec.DependencySpecifiers?.ToArray(), StatusCode = result.Result.StatusCode };
 
-                this.logger.LogWarning($"Received {(int)result.Result.StatusCode} {result.Result.ReasonPhrase} from {requestUri}. Waiting {timeSpan} before retry attempt {retryCount}");
+                this.logger.LogWarning(
+                    "Received {StatusCode} {ReasonPhrase} from {RequestUri}. Waiting {TimeSpan} before retry attempt {RetryCount}",
+                    (int)result.Result.StatusCode,
```

Comment:
Do we still need to manually cast?

```suggestion
                    (int)result.Result.StatusCode,
```